lib/rbi/rewriters/translate_rbs_sigs.rb

```diff
@@ -0,0 +1,91 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RBI
+  module Rewriters
+    # Translate all RBS signature comments to Sorbet RBI signatures
+    class TranslateRBSSigs < Visitor
+      extend T::Sig
+
+      class Error < RBI::Error; end
+
+      # @override
+      #: (Node? node) -> void
+      def visit(node)
+        return unless node
+
+        case node
+        when Tree
+          visit_all(node.nodes)
+        when AttrAccessor, AttrReader, AttrWriter
+          rbs_comments = extract_rbs_comments(node)
+          rbs_comments.each do |comment|
+            node.sigs << translate_rbs_attr_type(node, comment)
+          end
+        when Method
+          rbs_comments = extract_rbs_comments(node)
+          rbs_comments.each do |comment|
+            node.sigs << translate_rbs_method_type(node, comment)
+          end
+        end
+      end
+
+      private
+
+      #: (Method | Attr) -> Array[RBSComment]
+      def extract_rbs_comments(node)
+        comments = node.comments.dup
+        node.comments.clear
+
+        rbs_sigs = T.let([], T::Array[RBSComment])
+
+        comments.each do |comment|
+          case comment
+          when RBSComment
+            rbs_sigs << comment
+          else
+            node.comments << comment
+          end
+        end
+
+        rbs_sigs
+      end
```

Comment:
In line with my suggestion above, this could be:
```suggestion
      def transform_rbs_comments(node, &blk)
        comments = node.comments.dup
        node.comments.clear

        comments.each do |comment|
          case comment
          when RBSComment
            node.sigs << blk.call(comment)
          else
            node.comments << comment
          end
        end
      end
```